dart/test/sentry_client_test.dart

```diff
@@ -1458,6 +1538,21 @@ FutureOr<SentryEvent?> beforeSendCallback(SentryEvent event, {Hint? hint}) {
     ..sdk!.addPackage('test-pkg', '1.0');
 }
 
+FutureOr<SentryTransaction?> beforeSendTransactionCallback(
+    SentryTransaction transaction) {
+  return transaction
+    ..tags!.addAll({'theme': 'material'})
+    ..extra!['host'] = '0.0.0.1';
+
+  // TODO Check why event properties are dropped (constructor/copy). Is this expected?
```

Comment:
Could be a bug, has to be checked, did you try to use `copyWIth` instead?

Reply:
I fixed those issues. There were 2 properties that were never initialized if we don't pass it on through ctor. I'm not sure if they are relevant for the transaction and i didn't check if there are other params that need to be passed as well.